core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```diff
@@ -77,7 +80,7 @@
 
     private Settings settings = EMPTY_SETTINGS;
 
-    private Map<String, String> mappings = new HashMap<>();
+    private Map<String, Tuple<XContentType, BytesReference>> mappings = new HashMap<>();
```

Comment:
agreed. anyways we only support json and yaml as the methods deal with strings, yaml will be converted to json.